mcstatus/status_response.py

```diff
@@ -0,0 +1,326 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import Any, Dict, List, Optional, Union
+
+STYLE_MAP = {
+    "bold": "l",
+    "italic": "o",
+    "underlined": "n",
+    "obfuscated": "k",
+    "color": {
+        "dark_red": "4",
+        "red": "c",
+        "gold": "6",
+        "yellow": "e",
+        "dark_green": "2",
+        "green": "a",
+        "aqua": "b",
+        "dark_aqua": "3",
+        "dark_blue": "1",
+        "blue": "9",
+        "light_purple": "d",
+        "dark_purple": "5",
+        "white": "f",
+        "gray": "7",
+        "dark_gray": "8",
+        "black": "0",
+    },
+}
+
+
+@dataclass
+class AbstractDataclass(ABC):
+    """Abstract base class for all dataclasses.
+
+    We can't use just `ABC` because we need raise TypeError if we try to
+    instantiate an abstract dataclass. Class which inherit from `ABC` raises
+    it only if it has abstract methods.
+    """
+
+    def __new__(cls, *args, **kwargs):
+        if cls == AbstractDataclass or cls.__bases__[0] == AbstractDataclass:
+            raise TypeError("Cannot instantiate abstract class.")
+        return super().__new__(cls)
+
+
+@dataclass
+class MCServerResponse(AbstractDataclass):
```

Comment:
This class is purely for responses from the status protocol, not for query. It should therefore include `Status` somewhere in the class name, instead of just being `MCServerResponse`.

I wouldn't mind making a completely shared response class that would include query, however that's probably something for another PR since it would get way too overwhelming and for now, that's not what's happening here, so this class should be named accordingly.

This may also apply for other classes, I haven't looked into their naming too much.

Reply:
Fixed in 2180c13973dce05567ad758f261ad1b56dc027da.